common/common.h

```diff
@@ -79,6 +79,9 @@ struct gpt_params {
     float   yarn_beta_slow        = 1.0f;  // YaRN high correction dim
     int32_t yarn_orig_ctx         = 0;     // YaRN original context length
     float   defrag_thold          = -1.0f; // KV cache defragmentation threshold
+    std::vector<std::string> context_files;// context files to embed
+    int32_t chunk_size            = 64;    // chunk size for context embedding
+    std::string chunk_separator   = "\n";  // chunk separator for context embedding
```

Comment:
These params can be moved into `retrieval.cpp`, you can make a new struct `retrieval_params` to store them (again, just like in `quantize.cpp`)

https://github.com/ggerganov/llama.cpp/blob/1b26aebe4de4f048ac99996efd8a2c9af150904d/examples/quantize/quantize.cpp#L13

Reply:
@ngxson It took some more effort since unlike `quantize.cpp`, `retrieval` uses common gpt parameters like `--model`, `--top-k`, etc but I got it to work: [56b7db9](https://github.com/ggerganov/llama.cpp/pull/6193/commits/56b7db971e1851964f026a9977d3fbd2e7f46ca1) 🙂